From 4a67b147ad8550c6d1908a3ca971a0f410663298 Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Thu, 10 Mar 2022 08:57:51 -0800
Subject: Optimize single node kind cluster (#37862)

* Optimize single node kind cluster

Recommended by https://github.com/kubernetes/minikube/issues/5682

* Update prow/config/default.yaml

Co-authored-by: Eric Van Norman <ericvn@us.ibm.com>

Co-authored-by: Eric Van Norman <ericvn@us.ibm.com>
---
 prow/config/default.yaml | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/prow/config/default.yaml b/prow/config/default.yaml
index 1f2453aa7a..a6efb6fbfd 100644
--- a/prow/config/default.yaml
+++ b/prow/config/default.yaml
@@ -16,6 +16,11 @@ kubeadmConfigPatches:
       local:
         # Run etcd in a tmpfs (in RAM) for performance improvements
         dataDir: /tmp/kind-cluster-etcd
+    # We run single node, drop leader election to reduce overhead
+    controllerManagerExtraArgs:
+      leader-elect: "false"
+    schedulerExtraArgs:
+      leader-elect: "false"
     apiServer:
       extraArgs:
         "service-account-issuer": "kubernetes.default.svc"
-- 
2.35.3

